pantos/client/library/api.py

```diff
@@ -69,15 +68,15 @@ def decrypt_private_key(blockchain: Blockchain, keystore: str,
         If the private key cannot be loaded from the keystore file.
 
     """
-    _initialize_library()
+    _initialize_library(False)
     return _get_blockchain_client(blockchain).decrypt_private_key(
         keystore, password)
 
 
 def retrieve_service_node_bids(
         source_blockchain: Blockchain, destination_blockchain: Blockchain,
-        return_fee_in_main_unit: bool = True) \
-            -> _typing.Dict[BlockchainAddress, _typing.List[ServiceNodeBid]]:
+        return_fee_in_main_unit: bool = True, *, mainnet: bool = False) \
```

Comment:
Currently, the client library works only for one environment at a time (which was fine since only the testnet environment is public). In the future, it must work for both the mainnet and testnet environment without having to change the configuration (because external developers will use both of them). There will also be differences in the code since the protcol version in testnet will be different from the one in mainnet.

Reply:
I am concerned that this approach is going to be introducing more work than it should. By using the same code base to handle two environments, where the version of the testnet smart contracts >= mainnet smart contracts, we 
will have to add/change code based on new features.

Once those new features are ready and we and the devs/users are happy with it, we will have to clean up the code from old code, which is not working or used anymore.

Sounds like feature flags to me.

Wouldn't it be easier to handle it on tag or release level? Something like a pre-release?

Reply:
We actually discussed this already and decided that we don't want to have separate releases for different environments for the client library/CLI. With properly managing the protocol version (what we are implementing right now), I also don't see any issue here. We have clearly supported protocol versions from now on and can distinguish in the code where necessary. Code changes would also be necessary for separate releases, just that we have to manage them in separate branches, which doesn't make it easier. But I don't think that this even matters. An external developer will often want to develop for both environments and should not have to manage separate releases himself. We should make it as easy as possible to develop apps on top of Pantos.